src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/Regex.Count.cs

```diff
@@ -38,11 +38,10 @@ public int Count(ReadOnlySpan<char> input)
         {
             int count = 0;
 
-            RunAllMatchesWithCallback(input, 0, ref count, static (ref int count, Match match) =>
+            foreach (ValueMatch _ in EnumerateMatches(input))
```

Comment:
Assuming you keep this implememtation, using MoveNext directly will likely be a tad cheaper, as you don't need to access Current at all.